simulations/src/node/carnot/mod.rs

```diff
@@ -381,8 +381,21 @@ impl<L: UpdateableLeaderSelection, O: Overlay<LeaderSelection = L>> CarnotNode<O
                     parent_block=%block.parent(),
                     "receive approve message"
                 );
-                let (new, out) = self.engine.approve_block(block);
+                let block_grandparent_view = match &block.parent_qc {
+                    Qc::Standard(qc) => qc.view,
+                    Qc::Aggregated(_qc) => self
+                        .engine
+                        .safe_blocks()
+                        .iter()
+                        .filter(|(_, b)| matches!(b.parent_qc, Qc::Standard(_)))
+                        .max_by_key(|(_, b)| b.view)
+                        .map(|(_, b)| b.view)
+                        .unwrap_or_else(|| View::new(0)),
+                } - View::new(3);
```

Comment:
@Zeegomo , my logic here is to use the view of the most recent approved block that doesn't come from a new view (`Qc::Standard`). Am I missing something? 🤔